cypress/e2e/smoke/address_book.cy.js

```diff
@@ -20,91 +20,94 @@ const GNO_CSV_ENTRY = {
-      cy.get('[name="address"]').type(ADDRESS)
-      cy.get('[type="submit"]').click()
-
-      cy.findByText(NAME).should('exist')
-      cy.findByText(ADDRESS).should('exist')
+      cy.contains('Create entry').click()
+      cy.get('input[name="name"]').type(NAME)
+      cy.get('input[name="address"]').type(ENS_NAME)
+      // cy.wait(5000)
+      cy.contains('Save').click()
+
+      cy.contains(NAME).should('exist')
+      cy.contains(ADDRESS).should('exist')
     })
 
-    it('should edit an entry', () => {
+    it('should save an edited entry name', () => {
       // Click the edit button in the first entry
-      cy.findByTestId('EditIcon')
-        // <div> is not visible because its parent has CSS property: visibility: hidden
-        // so we have to use {force: true}
-        .click({ force: true })
-
-      cy.get('[name="name"]').clear().type(EDITED_NAME)
-      cy.get('[type="submit"]').click()
-      cy.findByText(NAME).should('not.exist')
-      cy.findByText(EDITED_NAME).should('exist')
+      cy.get('button[aria-label="Edit entry"]').click({ force: true })
+
+      // Give the entry a new name
+      cy.get('input[name="name"]').clear().type(EDITED_NAME)
+      cy.contains('Save').click()
+
+      // Previous name should have been replaced by the edited one
+      cy.get(NAME).should('not.exist')
+      cy.contains(EDITED_NAME).should('exist')
     })
 
     it('should delete an entry', () => {
       // Click the delete button in the first entry
-      cy.findByTestId('DeleteOutlineIcon').click({ force: true })
-      cy.findByText('Delete').should('exist').click()
-      cy.findByText(EDITED_NAME).should('not.exist')
+      cy.get('button[aria-label="Delete entry"]').click({ force: true })
+
+      cy.get('.MuiDialogActions-root').contains('Delete').click()
+      cy.get(EDITED_NAME).should('not.exist')
     })
   })
 
   describe('should import and export address book files', () => {
     it('should import an address book csv file', () => {
-      // Import CSV
-      cy.findByText('Import').click()
+      cy.contains('Import').click()
       cy.get('[type="file"]').attachFile('../fixtures/address_book_test.csv')
-      //This is the submit button for the Import modal. It requires an actuall class or testId to differentiate
-      //from the Import button at the top of the AB table
-      cy.contains('button.MuiButton-sizeMedium', 'Import').click()
-      cy.findByText(RINKEBY_CSV_ENTRY.name).should('exist')
-      cy.findByText(RINKEBY_CSV_ENTRY.address).should('exist')
-    })
 
-    it.skip('should find Gnosis Chain imported address', () => {
-      //Test skipped because it causes an error in the simulation
-      //"Unhandled Runtime Error
-      //Error: Hydration failed because the initial UI does not match what was rendered on the server."
+      // Import button should be enabled
+      cy.get('.MuiDialogActions-root').contains('Import').should('not.be.disabled')
```

Comment:
Can we instead of this class use something more generic, e.g. `button`?

Reply:
There is another button with 'Import' to import the _.csv_ that's why I needed to get more specific

Reply:
I assumed there will be only one Modal window open at a time. Felt more resilient than selecting the 2nd 'Import' button